datafusion/optimizer/src/type_coercion.rs

```diff
@@ -735,4 +786,27 @@ mod test {
             ),
         }))
     }
+
+    #[test]
+    fn test_type_coercion_rewrite() -> Result<()> {
+        let schema = Arc::new(
+            DFSchema::new_with_metadata(
+                vec![DFField::new(None, "a", DataType::Int64, true)],
+                std::collections::HashMap::new(),
+            )
+            .unwrap(),
+        );
+        let mut rewriter = TypeCoercionRewriter::new(schema);
+        let expr = is_true(
+            lit(ScalarValue::Int32(Some(12))).eq(lit(ScalarValue::Int64(Some(13)))),
+        );
```

Comment:
I wonder if you can write these tests like this to make them easier to read?

```suggestion
        let expr = is_true(
            lit(12i32).eq(lit(12i64),
        );
```

Reply:
`lit(12i32).eq(lit(13i64)),`